src/main/java/com/aws/greengrass/mqtt/bridge/clients/MQTTClient.java

```diff
@@ -163,8 +163,15 @@ public void start() throws MessageClientException {
     }
 
     private void disconnect() {
-        // 0ms quiescence time, just send the disconnect packet immediately
-        disconnect(0);
+        if (!mqttClientInternal.isConnected()) {
```

Comment:
Races with call to close? What if it isn't connected but it gets connected immediately after this?

Reply:
ah yeah thanks for the catch

Reply:
ran into null pointers with this change.  happens when force disconnect is called after close.  wouldn't expect this to happen with https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/pull/163, but i added a catch clause for this anyway for completeness, as awful as it is :) 
```
 java.lang.NullPointerException
	at org.eclipse.paho.client.mqttv3.internal.ClientComms.shutdownConnection(ClientComms.java:390)
	at org.eclipse.paho.client.mqttv3.internal.ClientComms.disconnectForcibly(ClientComms.java:559)
	at org.eclipse.paho.client.mqttv3.internal.ClientComms.disconnectForcibly(ClientComms.java:528)
	at org.eclipse.paho.client.mqttv3.MqttAsyncClient.disconnectForcibly(MqttAsyncClient.java:857)
	at org.eclipse.paho.client.mqttv3.MqttClient.disconnectForcibly(MqttClient.java:381)
	at com.aws.greengrass.mqtt.bridge.clients.MQTTClient.disconnect(MQTTClient.java:168)
	at com.aws.greengrass.mqtt.bridge.clients.MQTTClient.stop(MQTTClient.java:204)
	at com.aws.greengrass.mqtt.bridge.MQTTBridge.shutdown(MQTTBridge.java:165)
```
causes this, but that's fine

Reply:
does calling close on the client do a forceful disconnect? Can we just call close without calling disconnect during shutdown? Given that we're trying to do a force disconnect anyway.

Reply:
it's a little different in impl, even w/ a force close, it'll throw if the client is connected or connecting. and do nothing if disconnect is in progress

Reply:
but this force disconnect also fails if it is not connected or currently connecting? So basically we have no way to stop it while it is connecting? We must instead wait for it to complete the connection?

Reply:
force disconnect does not have the same behavior, doesn't make those same checks.  the way to stop while connecting would be disconnectForcibly

Reply:
ok that's good. I see the code no longer does a check for if you're connected or not, so that seems fine then.